### PR TITLE
feat: test and verify package on build

### PR DESCRIPTION
**Pull request by bruuuuuuuce**
follow-up to https://github.com/momentohq/client-sdk-javascript/pull/1045

Makes sure that nodejs and web sdk can be packaged and installed in our example repos, ensuring that the packing is correct

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -276,6 +276,51 @@ jobs:
         if: ${{ steps.validation.outcome == 'failure' }}
         run: exit 1
 
+  verify-packages:
+    strategy:
+      matrix:
+        node: [ 18 ]
+      fail-fast: true
+      max-parallel: 1
+    name: Test Deno examples on node ${{ matrix.node }}
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:285]:
```suggestion
    name: Verify packages are installable on node ${{ matrix.node }}
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -276,6 +276,56 @@ jobs:
...
+          popd
+          pushd packages/client-sdk-web
+            npm ci
+            npm run build
+            npm pack
+          popd
+
+      - name: Locally install in example and verify it builds
+        run: |
+          pushd example/nodejs/cache
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:317]:
```suggestion
          pushd examples/nodejs/cache
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -276,6 +276,56 @@ jobs:
...
+
+      - name: Locally install in example and verify it builds
+        run: |
+          pushd examples/nodejs/cache
+            npm i ../../packages/core/gomomento-sdk-core-0.0.1.tgz
+            npm i ../../packages/client-sdk-nodejs/gomomento-sdk-0.0.1.tgz
+            npm i
+            npm run build
+          popd
+          pushd example/web/cache
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:323]:
```suggestion
          pushd examples/web/cache
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -276,6 +276,56 @@ jobs:
...
+          pushd packages/client-sdk-web
+            npm ci
+            npm run build
+            npm pack
+          popd
+
+      - name: Locally install in example and verify it builds
+        run: |
+          pushd examples/nodejs/cache
+            npm i ../../packages/core/gomomento-sdk-core-0.0.1.tgz
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:318]:
```suggestion
            npm i ../../../packages/core/gomomento-sdk-core-0.0.1.tgz
```

Comment (bruuuuuuuce) [inline .github/workflows/build.yml:318]:
```suggestion
            npm i ../../../packages/core/gomomento-sdk-core-0.0.1.tgz
```

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -276,6 +276,56 @@ jobs:
...
+            npm i ../../packages/client-sdk-nodejs/gomomento-sdk-0.0.1.tgz
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:319]:
```suggestion
            npm i ../../../packages/client-sdk-nodejs/gomomento-sdk-0.0.1.tgz
```

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -276,6 +276,56 @@ jobs:
...
+            npm i
+            npm run build
+          popd
+          pushd examples/web/cache
+            npm i ../../packages/core/gomomento-sdk-core-0.0.1.tgz
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:324]:
```suggestion
            npm i ../../../packages/core/gomomento-sdk-core-0.0.1.tgz
```

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -276,6 +276,56 @@ jobs:
...
+            npm i ../../packages/client-sdk-web/gomomento-sdk-web-0.0.1.tgz
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:325]:
```suggestion
            npm i ../../../packages/client-sdk-web/gomomento-sdk-web-0.0.1.tgz
```

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -276,6 +276,56 @@ jobs:
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v3
+
+      - name: Build all packages
+        run: |
+          pushd packages/core
+            npm ci
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:295]:
```suggestion
            npm install
```

File: .github/workflows/build.yml [hunk 9]
```diff
@@ -276,6 +276,56 @@ jobs:
...
+            npm run build
+            npm pack
+          popd
+          pushd packages/common-integration-tests
+            npm ci
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:300]:
```suggestion
            npm install
```

File: .github/workflows/build.yml [hunk 10]
```diff
@@ -276,6 +276,56 @@ jobs:
...
+            npm run build
+            npm pack
+          popd
+          pushd packages/client-sdk-nodejs
+            npm ci
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:305]:
```suggestion
            npm install
```

File: .github/workflows/build.yml [hunk 11]
```diff
@@ -276,6 +276,56 @@ jobs:
...
+            npm run build
+            npm pack
+          popd
+          pushd packages/client-sdk-web
+            npm ci
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:310]:
```suggestion
            npm install
```

**Review by eaddingtonwhite — APPROVED** [review]
Nice work on this! Looks good should help us prevent merging any malformed packages! I think makes sense to just test this end to end vs try and verify package structure by hand. Hopefully will catch other future issues as well! I think it would be a good idea to add package/install verify step like this for other SDK's where makes sense as well. Maybe open some issues or add to notion page for CICD checks we have across SDK repos.

File: .github/workflows/build.yml [hunk 12]
```diff
@@ -276,6 +276,68 @@ jobs:
...
+            # This is a bit of a hack, but ensures that the package-lock does not get corrupted
+            # due to installing these locally packaged files. All we care about is that the examples
+            # use the locally packaged sdks, and that they can build correctly with no errors
+            rm -rf node_modules package-lock.json
+            npm i ../../../packages/core/gomomento-sdk-core-0.0.1.tgz
+            rm -rf node_modules package-lock.json
+            npm i ../../../packages/client-sdk-nodejs/gomomento-sdk-0.0.1.tgz
+            rm -rf node_modules package-lock.json
+            npm i
+            npm run build
```
Comment (eaddingtonwhite) [inline .github/workflows/build.yml:327]: @bruuuuuuuce maybe think about in follow up pr or issue be good to maybe try running example or run some test also make sure can actually start up? At least making sure builds for now is better then nothing here.

**Pull request merged**